docs/library-usage.md

```diff
@@ -0,0 +1,25 @@
+# Cloud SQL proxy dialer for Go
```

Comment:
IMO, this should stay in the README under the proxy module.

Reply:
Happy to do either -- just want to be consistent with the socket factory.